YelpAPITests/Client/YLPAutocompleteClientTestCase.m

```diff
@@ -0,0 +1,183 @@
+
+    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *request) {
+        return [request.URL.host isEqualToString:kYLPAPIHost];
+    } withStubResponse:^OHHTTPStubsResponse*(NSURLRequest *request) {
+        return [OHHTTPStubsResponse responseWithFileAtPath:OHPathForFile(self.defaultResource, self.class) statusCode:200 headers:@{@"Content-Type":@"application/json"}];
+    }];
+    
+    NSDictionary *expectedResponse = [self loadExpectedResponse:self.defaultResource];
+    NSDictionary *expectedFirstBiz = expectedResponse[@"businesses"][0];
+    NSDictionary *expectedSecondBiz = expectedResponse[@"businesses"][1];
+    NSDictionary *expectedThirdBiz = expectedResponse[@"businesses"][2];
+
+    double latitude = 37.7613570;
+    double longitude = -122.4243770;
+    
+    YLPCoordinate *myCoordinate = [[YLPCoordinate alloc] initWithLatitude:latitude longitude:longitude];
+    
+    [self.client fetchAutocompleteSuggestionsWithTerm:@"Bakery" coordinate:myCoordinate locale:nil completionHandler:^(YLPAutocomplete * _Nullable autocomplete, NSError * _Nullable error) {
+        
+        XCTAssertNil(error);
+        
+        NSMutableArray *businesses = [[NSMutableArray alloc] init];
+        
+        for (NSDictionary *dict in autocomplete.businesses) {
+            [businesses addObject:dict];
+        }
+        
+        XCTAssertEqualObjects([businesses[0] identifier], expectedFirstBiz[@"id"]);
+        XCTAssertEqualObjects([businesses[1] identifier], expectedSecondBiz[@"id"]);
+        XCTAssertEqualObjects([businesses[2] identifier], expectedThirdBiz[@"id"]);
+        
+        XCTAssertEqualObjects([businesses[0] name], expectedFirstBiz[@"name"]);
+        XCTAssertEqualObjects([businesses[1] name], expectedSecondBiz[@"name"]);
+        XCTAssertEqualObjects([businesses[2] name], expectedThirdBiz[@"name"]);
+
+        
+        [expectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:5 handler:nil];
+}
+
+- (void) testFetchAutocompleteResultsForCategories {
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Autocomplete query test, success case."];
+    
+    [OHHTTPStubs stubRequestsPassingTest:^BOOL(NSURLRequest *request) {
+        return [request.URL.host isEqualToString:kYLPAPIHost];
+    } withStubResponse:^OHHTTPStubsResponse*(NSURLRequest *request) {
+        return [OHHTTPStubsResponse responseWithFileAtPath:OHPathForFile(self.defaultResource, self.class) statusCode:200 headers:@{@"Content-Type":@"application/json"}];
+    }];
+    
+    NSDictionary *expectedResponse = [self loadExpectedResponse:self.defaultResource];
+    NSDictionary *expectedFirstCategory = expectedResponse[@"categories"][0];
+    NSDictionary *expectedSecondCategory = expectedResponse[@"categories"][1];
+    NSDictionary *expectedThirdCategory = expectedResponse[@"categories"][2];
+    
+    double latitude = 37.7613570;
```

Comment:
Since this isn't a required parameter for a category autocomplete search can we also add a test case for when latitude/longitude aren't provided. (once the code supports it)